lib/galaxy/jobs/__init__.py

```diff
@@ -1621,10 +1621,12 @@ def setup_external_metadata( self, exec_dir=None, tmp_dir=None,
         command = base_command
         if resolve_metadata_dependencies:
             metadata_tool = self.app.toolbox.get_tool("__SET_METADATA__")
-            dependency_shell_commands = metadata_tool.build_dependency_shell_commands(job_directory=self.working_directory)
-            if dependency_shell_commands:
-                dependency_shell_commands = "; ".join(dependency_shell_commands)
-                command = "%s; %s" % (dependency_shell_commands, command)
+            if metadata_tool is not None:
+                # Due to tool shed hacks for migrate and installed tool tests...
```

Comment:
I am :+1: on the change - however I would like this comment to point to the 'hack' it references.